src/UniversalTelegramBot.cpp

```diff
@@ -658,90 +669,88 @@ String UniversalTelegramBot::sendPostPhoto(JsonObject &payload) {
 }
 
 String UniversalTelegramBot::sendPhotoByBinary(
-    String chat_id, String contentType, int fileSize,
+    const String& chat_id, const String& contentType, int fileSize,
     MoreDataAvailable moreDataAvailableCallback,
-    GetNextByte getNextByteCallback) {
+    GetNextByte getNextByteCallback, GetNextBuffer getNextBufferCallback, GetNextBufferLen getNextBufferLenCallback) {
 
-  if (_debug)
-    Serial.println("SEND Photo");
+  #ifdef TELEGRAM_DEBUG  
+    Serial.println(F("sendPhotoByBinary: SEND Photo"));
+  #endif
 
-  String response = sendMultipartFormDataToTelegram(
-      "sendPhoto", "photo", "img.jpg", contentType, chat_id, fileSize,
-      moreDataAvailableCallback, getNextByteCallback);
+  String response = sendMultipartFormDataToTelegram("sendPhoto", "photo", "img.jpg",
+    contentType, chat_id, fileSize,
+    moreDataAvailableCallback, getNextByteCallback, getNextBufferCallback, getNextBufferLenCallback);
 
-  if (_debug)
+  #ifdef TELEGRAM_DEBUG  
     Serial.println(response);
+  #endif
 
   return response;
 }
 
-String UniversalTelegramBot::sendPhoto(String chat_id, String photo,
-                                       String caption,
+String UniversalTelegramBot::sendPhoto(const String& chat_id, const String& photo,
+                                       const String& caption,
                                        bool disable_notification,
                                        int reply_to_message_id,
-                                       String keyboard) {
-
-  DynamicJsonBuffer jsonBuffer;
-  JsonObject &payload = jsonBuffer.createObject();
+                                       const String& keyboard) {
 
+  DynamicJsonDocument payload(maxMessageLength);
   payload["chat_id"] = chat_id;
   payload["photo"] = photo;
 
-  if (caption) {
-    payload["caption"] = caption;
-  }
+  if (!caption.isEmpty())
```

Comment:
This is working for me? I also don't see it on the Arduino String docs

https://www.arduino.cc/reference/en/language/variables/data-types/stringobject/